proposals/0006-Keyboard-Friendly-Virtualization.md

```diff
@@ -0,0 +1,185 @@
+windows, or separate high-level APIs.
+
+### Endpoint-specific default behaviors
+
+VirtualizedList should be keyboard-friendly by default on desktop platforms as
+they are oriented towards usage of a physical keyboard. This additional
+realization is not free however, increasing memory usage to retain additional
+cells. This tradeoff may not make sense for mobile.
+
+This leads to desired behavior of:
+- **Keep FlatList focus-safe on desktop:** Common realization windows for
+needed for keyboard navigation are enabled by default.
+- **Keep FlatList memory usage steady on mobile:** New realization windows
+are opt-in where not pay to play.
+
+The full matrix of behaviors is described in the *API Reference* below. In
+general, **desktop users should realize **
+
+Endpoint detection is generally performed via `Platform` APIs, such as
+`Platform.isTV()` or `Platform.isPad()`. This change would add a
+`Platform.isDesktop()` endpoint query to allow desktop-specific behavior.
+
+### New VirtualizedList props
+
+Properties should be added to VirtualizedList (and FlatList/SectionList) to
+allow optimizing for app-specific needs. This can be exposed as a
+`RealizationWindowConfig`, describing the size and enablement of predefined
+realization windows. This API could be expanded in the future to allow
+dynamically controlled realization behaviors.
+
+
+### API reference
+
+```tsx
+/**
+ * Allows enabling or disabling an area for realization, optionally overriding
+ * its window size.
+ */
+export type RealizationWindow =
+  | boolean
+  | [boolean, {windowSize?: number}];
+
+/**
+ * Determines what areas of the list to keep realized. Enabling realization may
+ * be required for correct behavior when focus is out of viewport, but may also
+ * increase memory usage.
+ */
+export type RealizationWindowConfig = {
+  /**
+   * Keeps the area around the currently focused item realized.
+   * - Enabled by default on desktop/tv (where focus-loop is important)
+   * - Enabled by default on mobile, where this can only use more memory when
+   * physical-keyboard is used.
+   *
+   * Requires the default (CellRenderComponent)[
+   * https://reactnative.dev/docs/virtualizedlist#cellrenderercomponent] to be
+   * used
+   *
+   * Defaults to the (VirtualizedList windowSize)[
+   * https://reactnative.dev/docs/virtualizedlist#windowsize]
+   */
+  focused?: RealizationWindow;
+
+  /**
+   * Keeps the beginning area of the list realized.
+   * - Enabled by default on desktop (for home-key navigation)
+   * - Disabled by default on mobile/TV
+   *
+   * Defaults to **half** the (VirtualizedList windowSize)[
+   * https://reactnative.dev/docs/virtualizedlist#windowsize]
+   */
+  home?: RealizationWindow;
+
+  /**
+   * Keeps the end area of the list realized.
+   * - Enabled by default on desktop (for end-key navigation)
+   * - Disabled by default on mobile/TV
+   *
+   * Defaults to **half** the (VirtualizedList windowSize)[
```

Comment:
Half of windowsize means we have a full buffer at both ends, but also means we effectively double the number of realized cells on desktop. Think this should maybe be lower, but there are also some nice properties with current value.